### PR TITLE
Added make build-test-container and reorganized makefile

### PR DESCRIPTION
**Pull request by thom-at-redhat**
# Description

Main issue was to add target to build test container.
Added generalized build-containers target
Renamed build to build-edge-api-container
Alphabetized targets since we were missing help text
Added help text for missing targets
Ran through "checkmate" with many issues resolved

Fixes # (issue)

## Type of change

What is it?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [X] Refactor

# Checklist:

- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I run `make pre-commit` to check fmt/vet/lint/test-no-fdo

### CODE REVIEW

**Comment by InsightsDroid** [comment]
Test results are available in [Ibutsu](https://url.corp.redhat.com/ibutsu-runs). The test run IDs are:
cc041b4f-3db6-4dc9-903f-7c95f8a5af52

File: Makefile [hunk 1]
```diff
@@ -7,110 +7,127 @@ ifeq ($(UNAME_S),Darwin)
...
+bonfire-config-github:
+	@cp default_config.yaml.github.example config.yaml
 
-test:
-	go test $(BUILD_TAGS) $$(go list $(BUILD_TAGS) ./... | grep -v /test/) $(TEST_OPTIONS)
+build-containers: build-edge-api-container build-test-container
 
-test-no-fdo:
-	go test $$(go list ./... | grep -v /test/) $(TEST_OPTIONS)
+build-edge-api-container:
```
Comment (loadtheaccumulator) [inline Makefile:29]: Do we want to allow a variable for the :tag to be set explicitly, and do we want to default to latest?
We can also allow for a different Dockerfile|Containerfile.

Our build_deploy.sh script uses this...
podman build --no-cache -f Dockerfile -t "${IMAGE}:${IMAGE_TAG}" .

**Review by loadtheaccumulator — CHANGES_REQUESTED** [review]
question re: possible changes to container build command

File: Makefile (hunk 1 above)
Comment (thom-at-redhat) [reply Makefile:34]: Made changes to address but request re-review

File: Makefile [hunk 2]
```diff
@@ -7,110 +7,142 @@ ifeq ($(UNAME_S),Darwin)
...
-	@echo ""
+CONTAINERFILE_NAME=Dockerfile
 
+.PHONY: all build-containers clean help test
 
-test:
-	go test $(BUILD_TAGS) $$(go list $(BUILD_TAGS) ./... | grep -v /test/) $(TEST_OPTIONS)
+bonfire-config-local:
+	@cp default_config.yaml.local.example config.yaml
+	@sed -i ${OS_SED} 's|REPO|$(PWD)|g' config.yaml
```
[the remark this answers is not in the document]
Comment (thom-at-redhat) [reply Makefile]: Fixed

**Pull request merged**